hypothesis-python/src/hypothesis/extra/pytestplugin.py

```diff
@@ -179,8 +180,19 @@ def pytest_collection_modifyitems(items):
     for item in items:
         if not isinstance(item, pytest.Function):
             continue
-        if getattr(item.function, 'is_hypothesis_test', False):
+        if is_hypothesis_test(item.obj):
             item.add_marker('hypothesis')
+        if getattr(item.obj, 'is_hypothesis_strategy_function', False):
+            def note_strategy_is_not_test(*args, **kwargs):
+                note_deprecation(
+                    '%s is a function that returns a Hypothesis strategy, '
+                    'but Pytest has collected it as a test function.  This '
```

Comment:
nit: should "Pytest" really have an upper case p?

Reply:
🤦‍♂️ I *have a pytest commit bit*, I should know the style by now...